### DISCUSSION TITLE
Upgrade keeps hang on Switching to safeboot partition...

### DISCUSSION THREAD

**Discussion by kroon040** [arendst/Tasmota#21113] [Q&A] [answered]
Hi,

I want to upgrade my tasmota32c6 device, but when I do this, it keeps hanging on Switching to safeboot partition...

I have not OTA, filled in but only uploading trough the web interface. Show there be an image to go first to safeboot partition, or do I something else wrong. I can't get easy with the chip it is soldered in the device.

**Comment by sfromis** [comment]
What binary? I had no trouble doing OTA by file upload to a downloaded copy of `tasmota32c6cdc-arduino30.bin`. Anything in console?

The process is that the upgrade should quickly switch to the safeboot partition with the browser retrying when safeboot is available, to trigger the file transfer process.

**Reply by kroon040** [reply]
If i open 2 sessions in my browser one with the upload the other for console, than I see a reboot of tasmota , but no switchboot temporary , just back to the original not upgrading. Is uploading a that factory image possible, or do i break than it?

**Reply by barbudor** [reply]
Factory is ONLY for serial flashing. You haven't broken but Tasmota do not start upgrade wehn detecting wrong firmware
Use the non-factory bin for upgrade

**Reply by kroon040** [reply]
I have this as current info on the info page

program	13.4.0.2 (arduino30) 

ESP Chip Id	16776771 (ESP32-C6 v0.0)

Try to flash the new compiled image

tasmota32c6-arduino30.bin

So this should be correct I think

**Reply by kroon040** [reply]
Some more info

Flash Size	4096 KB
Program Flash Size	4096 KB
Program Size	2022 KB
Free Program Space	857 KB
Free Memory	290.5 KB (frag. 5%)
Partition safeboot	832 KB
Partition app0*	2880 KB (used 70%)
Partition fs	320 KB

Image size i want to  flash 
2,0M  6 apr 18:45 tasmota32c6-arduino30.bin
2072608  6 apr 18:45 tasmota32c6-arduino30.bin

**Comment by kroon040** [comment]
Hi,

I now try to update the 13.4.1.2 to 14.0. I noticed that I don't have the working safeboot. Do an restart 3, I goes back to original firmware.

I noticed when uploading the partitionwizard I see this, the safeboot used size is unknown. (see below) I there a way to get a good bin on it are a good safeboot to get it working again. I can't get it to flash serial again the bulb I had to glue when replacing the chip with an esp32c6

[safeboot]: 	832 KB 	 (unknown)
app0: 	2880 KB 	 (used 69%)
fs: 	320 KB

**Reply by alexgill** [reply, marked as answer]
Hi, maybe try upgrading your *safeboot* with the `upgrade 2` command, and once it completes, try your original upgrade.
>  Upgrade 	1 = download firmware from OtaUrl and restart
>  &emsp; &emsp; 2 = (ESP32 only) download safeboot firmware based on OtaUrl and restart into safeboot

**Reply by kroon040** [reply]
@alexgill , thnx firist the upgrade 2 worked to restore the safeboot, than the upgrade 1 , and now it looks fine again and on tasmota 14.
